libs/core/src/lib/notification/notification-group/notification-group.component.ts

```diff
@@ -1,14 +1,60 @@
-import { ChangeDetectionStrategy, Component, HostBinding, ViewEncapsulation } from '@angular/core';
+import { 
+    ChangeDetectionStrategy,
+    Component,
+    ElementRef,
+    HostBinding,
+    Input,
+    ViewEncapsulation,
+    OnChanges,
+    OnInit
+} from '@angular/core';
+import { applyCssClass, CssClassBuilder } from '../../utils/public_api';
 
 @Component({
     selector: 'fd-notification-group',
-    templateUrl: './notification-group.component.html',
-    styleUrls: ['./notification-group.component.scss'],
+    template: `<ng-content></ng-content>`,
     encapsulation: ViewEncapsulation.None,
     changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class NotificationGroupComponent {
+export class NotificationGroupComponent  implements OnChanges, OnInit, CssClassBuilder {
+    /** User's custom classes */
+    @Input()
+    class: string;
+
+    /** Whether the Notification Group is in mobile mode */
+    @Input() mobile = false;
+
+    /** User defined width for the notification */
+    @HostBinding('style.width')
+    @Input() width:  string;
+
+    /** @hidden */
+    constructor(private _elementRef: ElementRef) {}
+
+    /** @hidden */
+    ngOnChanges(): void {
+        this.buildComponentCssClass();
+    }
+
+    @applyCssClass
+    /** CssClassBuilder interface implementation
+     * function is responsible for order which css classes are applied
+     */
+    buildComponentCssClass(): string[] {
+        return [
+            'fd-notification fd-notification--group fd-notification-custom-block',
+            this.mobile ? 'fd-notification--mobile' : '',
+            this.class
+        ];
+    }
+
+    /** @hidden */
+    elementRef(): ElementRef<any> {
```

Comment:
Do we need an explicit type argument "any" here?

Reply:
yes. That's also how it's done in the other components:
<img width="373" alt="Screen Shot 2021-05-03 at 11 06 51 AM" src="https://user-images.githubusercontent.com/39598672/116894136-ba90e400-abff-11eb-9d6c-0904da8bd5c5.png">

Reply:
![image](https://user-images.githubusercontent.com/6586561/116923896-dd52e500-ac5f-11eb-9834-d9532efac4dd.png)
But the generic type of ElementRef is any by default, so this code is copying the default behavior

Reply:
ok...will remove it but it will be inconsistent with the rest of the app. Prob need to create an issue to refactor it for all components.